sources/api/sundog/src/main.rs

```diff
@@ -221,6 +224,54 @@ where
     Ok(populated_settings)
 }
 
+// Builds the proxy environment variables to pass to settings generators
+async fn build_proxy_env<P>(socket_path: P) -> Result<HashMap<String, String>>
+where
+    P: AsRef<Path>,
+{
+    // Retrieve network proxy-related settings.
+    let settings: serde_json::Value =
+        schnauzer::get_json(&socket_path, "/settings", Some(("prefix", "network")))
+            .await
+            .context(error::RetrieveSettingsSnafu)?;
+    let mut proxy_envs = HashMap::new();
+    if let Some(network_settings) = settings.get("network") {
+        if let Some(https_proxy) = network_settings.get("https-proxy").and_then(|s| s.as_str()) {
+            proxy_envs.insert("https_proxy".to_string(), https_proxy.to_string());
+            proxy_envs.insert("HTTPS_PROXY".to_string(), https_proxy.to_string());
+        }
+        if let Some(no_proxy) = network_settings.get("no-proxy").and_then(|v| v.as_array()) {
```

Comment:
nit: we could do an early return here since we don't care about `NO_PROXY` if we don't have either proxy settings.